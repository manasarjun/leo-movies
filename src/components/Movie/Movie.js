import React, { useContext } from 'react';
import {
  Card, CardActions,
  CardContent, CardMedia, IconButton, Typography,
} from '@material-ui/core';
import BookmarkIcon from '@material-ui/icons/Bookmark';
import FavoriteIcon from '@material-ui/icons/Favorite';
import { makeStyles } from '@material-ui/core/styles';

import { StoreContext, FAVOURITES, WATCHLIST } from '../../provider/Provider';

const useStyles = makeStyles({
  root: {
    maxWidth: 500,
    margin: '12px auto',
  },
  favourite: {
    color: (props) => (props.isFavourite ? 'red' : null),
  },
  watchlist: {
    color: (props) => (props.isWatchList ? 'red' : null),
  },
});


export default function Movie(props) {

  const classes = useStyles(props);
  const { movie, isFavouriteItem, isWatchItem, isTrending } = props;
  const Store = useContext(StoreContext);

  const handleFavourites = (mov) => {
    const storedFavourites = Store.favourites;
    const existedList = storedFavourites.filter(s => s.id === mov.id)
    if (existedList.length === 0) {
      const favourite = {
        id: mov.id,
        title: mov.title,
        overview: mov.overview,
      };
      Store.setFavourites(favourite, FAVOURITES);
    } else {
      Store.updateFavourites(mov.id, FAVOURITES);
    }
  };

  const handleWatchList = (mov) => {
    const storedWatchList = Store.watchList;
    const existedList = storedWatchList.filter(s => s.id === mov.id)
    if (existedList.length === 0) {
      const watchList = {
        id: mov.id,
        title: mov.title,
        overview: mov.overview,
      };
      Store.setWatchList(watchList, WATCHLIST);
    } else {
      Store.updateWatchList(mov.id, WATCHLIST);
    }
  };

  return (
    <Card className={classes.root}>
      {!isWatchItem && !isFavouriteItem ? <CardMedia
        className={classes.sectionDesktop}
        component="img"
        height="300"
        image={`https://image.tmdb.org/t/p/w500/${movie.poster_path}`}
      /> : null}
      <CardActions>
        <IconButton
          aria-label="add to favorites"
          onClick={() => handleFavourites(movie)}
        >
          {!isWatchItem ? <FavoriteIcon className={classes.favourite} /> : null}
        </IconButton>
        <IconButton
          onClick={() => handleWatchList(movie)}
        >
          {!isFavouriteItem ? <BookmarkIcon className={classes.watchlist} /> : null}
        </IconButton>
        <> {isTrending ?
          <Typography variant="body2" color="textSecondary" component="p">
            <p>Votes: {movie.vote_count}    Rating: {movie.vote_average}</p>
          </Typography> : null}</>
      </CardActions>
      <CardContent>

        <Typography variant="h5" component="h2">
          {movie.title}

        </Typography>

        <Typography variant="body2" color="textSecondary" component="p">
          {movie.overview}
        </Typography>
      </CardContent>
    </Card>
  );
}
