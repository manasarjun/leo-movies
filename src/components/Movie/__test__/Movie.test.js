import React from 'react';
import renderer from 'react-test-renderer';

import Movie from '../Movie';

describe('Movie', () => {
  const movie = {
    id: 123,
    title: 'avengers',
    overview: 'avengers',
    poster_path: '467.png',
  };

  const props = {
    isFavouriteItem: true,
    isWatchItem: true,
    isTrending: true,
    movie,
  };

  it('should render Movie', () => {
    const tree = renderer.create(<Movie {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
