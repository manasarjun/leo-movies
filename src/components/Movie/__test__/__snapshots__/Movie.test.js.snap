// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should render Movie 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      aria-label="add to favorites"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      />
    </button>
    <button
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      />
    </button>
     
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      Votes: 
          Rating: 
    </p>
  </div>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5"
    >
      avengers
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      avengers
    </p>
  </div>
</div>
`;
